kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -183,25 +180,42 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(Map<TopicPartition, Offs
   private PartitionedTableId getRecordTable(SinkRecord record) {
     String tableName;
     String dataset = config.getString(BigQuerySinkConfig.DEFAULT_DATASET_CONFIG);
-    String[] smtReplacement = record.topic().split(":");
-
-    if (smtReplacement.length == 2) {
-      dataset = smtReplacement[0];
-      tableName = smtReplacement[1];
-    } else if (smtReplacement.length == 1) {
-      tableName = smtReplacement[0];
+    if (topic2TableMap != null) {
+      // TODO what if the map doesn't contain the topic name?
+      // We don't need an explicit call to sanitize here because if the
+      // sanitize flag is true, then the topic2TableMap would already contain
+      // sanitized table names.
+      tableName = topic2TableMap.get(record.topic());
     } else {
-      throw new ConnectException(String.format(
-          "Incorrect regex replacement format in topic name '%s'. "
-              + "SMT replacement should either produce the <dataset>:<tableName> format " 
-              + "or just the <tableName> format.",
-          record.topic()
-      ));
-    }
+      String[] smtReplacement = record.topic().split(":");
```

Comment:
@sagarrao12 minor comment on the naming here. Why is the prefix smt used? I ACK that it was named so in the previous code to but I do not see it has anything to do with "single message transform". If that's correct can you rename it please?

Reply:
haha that's just due to copy/paste :) . Would remove it.

Reply:
Actually, it's because there's a provision in the connector to user regexrouter smt to re-set the topic and that's why you see those variable names.